tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/cli/dockerCli/core/dockerTaskEngine.ts

```diff
@@ -0,0 +1,265 @@
+import { dockerTaskEngineConfig } from "../schema/dockerTaskEngineConfig";
+import {
+    addFileLog,
+    CodegenToSdkConfig,
+    GenerateAndBuildInput,
+    GenerateAndBuildOptions,
+    getCodegenToSdkConfig,
+    getGenerateAndBuildOutput,
+    getTask,
+    getTestOutput,
+    InitOptions,
+    initOutput,
+    MockTestInput,
+    MockTestOptions,
+    removeFileLog,
+    requireJsonc,
+    runScript,
+    StringMap
+} from "@azure-tools/sdk-generation-lib";
+import * as path from "path";
+import * as fs from "fs";
+import { writeFileSync } from "fs";
+import { disableFileMode, getHeadRef, getHeadSha, safeDirectory } from "../../../utils/git";
+import { Logger } from "winston";
+import { execSync } from "child_process";
+import { DockerContext } from "./DockerContext";
+
+export type DockerTaskEngineContext = {
+    logger: Logger;
+    configFilePath: string;
+    initOutput: string;
+    generateAndBuildInputJson: string;
+    generateAndBuildOutputJson: string;
+    mockTestInputJson: string;
+    mockTestOutputJson: string;
+    initTaskLog: string;
+    generateAndBuildTaskLog: string;
+    mockTestTaskLog: string;
+    readmeMdPath: string;
+    specRepo: {
+        repoPath: string;
+        headSha: string;
+        headRef: string;
+        repoHttpsUrl: string;
+    };
+    serviceType?: string;
+    tag?: string;
+    sdkRepo: string;
+    resultOutputFolder?: string;
+    envs?: StringMap<string | boolean | number>;
+    packageFolders?: string[];
+    mockServerHost?: string;
+    taskResults?: {};
+    taskResultJsonPath: string;
+    changeOwner: boolean
+}
+
+export function initializeDockerTaskEngineContext(dockerContext: DockerContext): DockerTaskEngineContext {
```

Comment:
Can you put those functions into a class like DockerExecuteEngine?

Reply:
I warp them in `DockerTaskEngineContext`, together with necessary information. Then we can use the `DockerTaskEngineContext` for all task engine related work.